recipe/meta.yaml

```diff
@@ -0,0 +1,94 @@
+# Builds the CPU and CUDA variants of liblightgbm-base and py-lightgbm-base package
+{% set name = "lightgbm-base" %}
+{% set version = "3.1.0" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  git_url: https://github.com/microsoft/LightGBM
+  git_tag: v{{ version }}
+  patches:
+    - 0001-Build-fix-for-cuda.patch  #[build_type == 'cuda']
+
+build:
+  number: 1
+
+outputs:
+  - name: liblightgbm-base
+    script: install-liblightgbm.sh
+    build:
+      string: {{ build_type }}_{{ PKG_BUILDNUM }} #[build_type == 'cpu']
+      string: {{ build_type }}{{ cudatoolkit | replace(".*", "") }}_{{ PKG_BUILDNUM }} #[build_type == 'cuda']
+{% if build_type == 'cuda' %}
+      script_env:
+        - CUDA_HOME
+{% endif %}
+
+    requirements:
+      build:
+        - {{ compiler('c') }}
+        - {{ compiler('cxx') }}
+        - cmake >=3.16 
+        - make {{ make }}
+        - libuv
+      host:
+        - cudatoolkit {{ cudatoolkit }}        #[build_type == 'cuda']
+        - openmpi {{ openmpi }}                #[mpi_type == 'openmpi']
+      run:
+        - cudatoolkit {{ cudatoolkit }}        #[build_type == 'cuda']
+        - _lightgbm_select {{ lightgbm_select_version }} 
+
+  - name: py-lightgbm-base
+    script: install-py-lightgbm.sh
+    build:
+      string: {{ build_type }}_py{{ python | replace(".", "") }}_{{ PKG_BUILDNUM }} #[build_type == 'cpu']
```

Comment:
mpi_type should be added to the build strings.

Reply:
In case of None as mpi type,  build string will have that too in it. Will it be fine?